aspnet/fundamentals/request-features.rst

```diff
@@ -12,63 +12,61 @@ Individual web server features related to how HTTP requests and responses are ha
 
-`IHttpSendFileFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/IHttpSendFileFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.IHttpSendFileFeature`
   Defines a method for sending files asynchronously.
 
-`IHttpWebSocketFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/IHttpWebSocketFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.IHttpWebSocketFeature`
   Defines an API for supporting web sockets.
 
-`IHttpRequestIdentifierFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/IHttpRequestIdentifierFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.IHttpRequestIdentifierFeature`
   Adds a property that can be implemented to uniquely identify requests.
 
-`ISessionFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/ISessionFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.ISessionFeature`
   Defines ``ISessionFactory`` and ``ISession`` abstractions for supporting user sessions.
 
-`ITlsConnectionFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/ITlsConnectionFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.ITlsConnectionFeature`
   Defines an API for retrieving client certificates.
 
-`ITlsTokenBindingFeature <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/Features/ITlsTokenBindingFeature/index.html>`_
+:dn:iface:`~Microsoft.AspNetCore.Http.Features.ITlsTokenBindingFeature`
   Defines methods for working with TLS token binding parameters.
 
-.. note:: ``ISessionFeature`` is not a server feature, but is implemented by the ``SessionMiddleware`` (see :doc:`/fundamentals/app-state`).
+.. note:: :dn:iface:`~Microsoft.AspNetCore.Http.Features.ISessionFeature` is not a server feature, but is implemented by the :dn:cls:`~Microsoft.AspNetCore.Session.SessionMiddleware` (see :doc:`/fundamentals/app-state`).
   
 Feature collections
 -------------------
 
-The `HttpContext.Features <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/HttpContext/index.html#prop-Microsoft.AspNetCore.Http.HttpContext.Features>`_ property provides an interface for getting and setting the available HTTP features for the current request. Since the feature collection is mutable even within the context of a request middleware can be used to modify the collection and add support for additional features.
+The :dn:prop:`~Microsoft.AspNetCore.Http.HttpContext.Features` property of :dn:cls:`~Microsoft.AspNetCore.Http.HttpContext` provides an interface for getting and setting the available HTTP features for the current request. Since the feature collection is mutable even within the context of a request middleware can be used to modify the collection and add support for additional features.
```

Comment:
need comma after request
Since the feature collection is mutable even within the context of a request,